python/ray/train/batch_predictor.py

```diff
@@ -187,14 +197,26 @@ def __init__(self):
 
             def __call__(self, batch):
                 if feature_columns:
-                    prediction_batch = batch[feature_columns]
+                    if batch_format == "numpy":
+                        prediction_batch = {
```

Comment:
`batch` may be a single ndarray for the `"numpy"` batch format, so that case should be handled as well.